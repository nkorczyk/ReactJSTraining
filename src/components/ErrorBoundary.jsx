import React, { Component } from 'react';

class ErrorBoundary extends Component {
  state = {
    hasError: false,
    error: null,
    errorInfo: null
  };

  componentDidCatch(error, errorInfo) {
    this.setState({
      hasError: true,
      error,
      errorInfo
    });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div className="msg msg-error z-depth-3">
          Error occurred while fetching the data!
      </div>
      );
    }
    return this.props.children;
  }
}

export default ErrorBoundary;
