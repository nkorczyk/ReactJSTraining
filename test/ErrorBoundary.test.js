import { shallow } from 'enzyme';
import React from 'react';
import ErrorBoundary from '../src/components/ErrorBoundary';

describe('ErrorBoundary Component', () => {
  let element;
  beforeEach(() => {
    element = shallow(<ErrorBoundary><div>custom content</div></ErrorBoundary>);
  });

  test('Snapshot test with default props', () => {
    expect(element).toMatchSnapshot();
    expect(element.html()).toEqual('<div>custom content</div>');
  });

  test('Should render error message when hasError is set in the state', () => {
    element.setState({ hasError: true });
    expect(element.text()).toEqual('Error occurred while fetching the data!');
  });

  test('Should update component state when componentDidCatch is invoked', () => {
    element.instance().componentDidCatch('error', 'error');
    expect(element.state('hasError')).toEqual(true);
  });
});
