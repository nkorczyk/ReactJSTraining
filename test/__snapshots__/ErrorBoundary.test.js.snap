// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary Component Snapshot test with default props 1`] = `
<div>
  custom content
</div>
`;
